### PR TITLE
feat: sync v0.46.0 with changes in v0.44.x

### PR DESCRIPTION
**Pull request by vivshankar**
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: handler/rfc8693/access_token_type_handler.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+		}
+
+		response.SetExtra("refresh_token", refresh)
+	}
+
+	response.SetAccessToken(token)
+	response.SetTokenType("bearer")
+	response.SetExpiresIn(c.getExpiresIn(request, fosite.AccessToken, c.AccessTokenLifespan, time.Now().UTC()))
+	response.SetScopes(request.GetGrantedScopes())
+
```
Comment (asasmita) [inline handler/rfc8693/access_token_type_handler.go:189]: shall we add `issued_token_type` here

File: handler/rfc8693/custom_jwt_type_handler.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+	tokenTypes := c.Config.GetTokenTypes(ctx)
+	tokenType := tokenTypes[requestedTokenType]
+	if tokenType == nil || tokenType.GetType(ctx) != JWTTokenType {
+		return nil
+	}
+
+	if err := c.issue(ctx, request, tokenType, responder); err != nil {
+		return err
+	}
+
```
Comment (asasmita) [inline handler/rfc8693/custom_jwt_type_handler.go:93]: should we add `issued_token_type`here

File: handler/rfc8693/id_token_type_handler.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	}
+
+	token, err := c.IssueStrategy.GenerateIDToken(ctx, c.Config.GetIDTokenLifespan(ctx), request)
+	if err != nil {
+		return err
+	}
+
+	response.SetAccessToken(token)
+	response.SetTokenType("N_A")
+
```
Comment (asasmita) [inline handler/rfc8693/id_token_type_handler.go:146]: shall we add `issued_token_type`

File: handler/rfc8693/refresh_token_type_handler.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+				err = rollBackTxnErr
+			}
+			return errors.WithStack(fosite.ErrServerError.WithDebug(err.Error()))
+		}
+	}
+
+	response.SetAccessToken(refresh)
+	response.SetTokenType("N_A")
+	response.SetExpiresIn(c.getExpiresIn(request, fosite.RefreshToken, c.RefreshTokenLifespan, time.Now().UTC()))
+	response.SetScopes(request.GetGrantedScopes())
```
Comment (asasmita) [inline handler/rfc8693/refresh_token_type_handler.go:175]: should we add `issued_token_type`

File: handler/rfc8693/access_token_type_handler.go [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+	// Scope check
+	for _, scope := range request.GetRequestedScopes() {
+		if !c.ScopeStrategy(or.GetGrantedScopes(), scope) {
+			return nil, nil, errors.WithStack(fosite.ErrInvalidScope.WithHintf("The subject token is not granted \"%s\" and so this scope cannot be requested.", scope))
+		}
+	}
+
+	// Convert to flat session with only access token claims
+	tokenObject := session.AccessTokenClaimsMap()
```
Comment (asasmita) [inline handler/rfc8693/access_token_type_handler.go:147]: this one should use original session, not the session

File: handler/rfc8693/refresh_token_type_handler.go [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+	// Scope check
+	for _, scope := range request.GetRequestedScopes() {
+		if !c.ScopeStrategy(or.GetGrantedScopes(), scope) {
+			return nil, nil, errors.WithStack(fosite.ErrInvalidScope.WithHintf("The subject token is not granted \"%s\" and so this scope cannot be requested.", scope))
+		}
+	}
+
+	// Convert to flat session with only access token claims
+	tokenObject := session.AccessTokenClaimsMap()
```
Comment (asasmita) [inline handler/rfc8693/refresh_token_type_handler.go:148]: this one should use original session, not the session

**Comment by vivshankar** [comment]
@asasmita I will sync the token exchange changes into the fork here in another PR. The purpose of this PR is to make sure we have everything in the latest Fosite release pulled in. It appears we should be good to go, so I will merge this PR for now.

**Pull request merged**